component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```diff
@@ -260,30 +208,54 @@ public void stop() {
-    @ApiResponse (responseCode = "400", description = "Invalid query input") })
-  public Response getUsers(@Context UriInfo uriInfo,
-                           @Parameter(description = "User name information to filter, ex: user name, last name, first name or full name") @QueryParam("q") String q,
-                           @Parameter(description = "User status to filter online users, ex: online") @QueryParam("status") String status,
-                           @Parameter(description = "User type to filter, ex: internal, external") @DefaultValue("internal") @QueryParam("userType") String userType,
-                           @Parameter(description = "Is connected users") @QueryParam("isConnected") String isConnected,
-                           @Parameter(description = "Space id to filter only its members, ex: 1") @QueryParam("spaceId") String spaceId,
-                           @Parameter(description = "Is disabled users") @Schema(defaultValue = "false") @QueryParam("isDisabled") boolean isDisabled,
-                           @Parameter(description = "Enrollment status, ex: enrolled, not enrolled, no possible enrollment") @QueryParam("enrollmentStatus") String enrollmentStatus,
-                           @Parameter(description = "Offset") @Schema(defaultValue = "0") @QueryParam("offset") int offset,
-                           @Parameter(description = "Limit") @Schema(defaultValue = "20") @QueryParam("limit") int limit,
-                           @Parameter(description = "Returning the number of users found or not") @Schema(defaultValue = "false") @QueryParam("returnSize") boolean returnSize,
-                           @Parameter(description = "Asking for a full representation of a specific subresource if any") @QueryParam("expand") String expand) throws Exception {
+  @Operation(summary = "Gets all users", method = "GET", description = "Using the query param \"q\" to filter the target users, ex: \"q=jo*\" returns all the users beginning by \"jo\"."
+      + "Using the query param \"status\" to filter the target users, ex: \"status=online*\" returns the visible online users."
+      + "Using the query params \"status\" and \"spaceId\" together to filter the target users, ex: \"status=online*\" and \"spaceId=1*\" returns the visible online users who are member of space with id=1."
+      + "The params \"status\" and \"spaceId\" cannot be used with \"q\" param since it will falsify the \"limit\" param which is 20 by default. If these 3 parameters are used together, the parameter \"q\" will be ignored")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "404", description = "Resource not found"),
+      @ApiResponse(responseCode = "500", description = "Internal server error due to data encoding"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input") })
```

Comment:
why changing the code format that was well displayed here ? Please try to format the changed code only (Best Practice Rule) to make you change clear and readable.

Reply:
done